### ISSUE TITLE
Enhancement Request - beyond draft/final output modes, add more

### ISSUE THREAD

**Issue by bwgartner** [openSUSE/daps#638]
### Problem description
Beyond the binary states of DRAFT=yes, requesting the ability to put other designations, mapping into the lifecycle that the document is in. For instance, would almost be nice to match things like Experimental, Alpha, Beta, RC (release candidate) and extend beyond final to Archive/Retired.

### Expected behavior
Allow other STATE=yes modes with the designated state word to be in the compiled/published format (like Draft currently does). Bonus points for also doing some potential, respective color mapping (within the branding scheme provided by CSS) to also trigger a designation for the readers.

And fair game also to maybe link to the ROLE/PROFROLE designations as well.

### Steps to reproduce problem
Currently only aware of DRAFT=yes or commented out, when looking at the codebase. Apologies if I missed other options

**Comment by fsundermeyer** [comment]
You are right, we currently have DRAFT as the only option to declare the state of the document.

Apart from some color mapping, how would you like to have the state visualized in the docs? A watermak background image like we have for DRAFT? 
Shopuld the state also be taken into account on the navigation pages of the web server?
If you want any of this to be implemented we need to define a fixed list of states we would support.

In case you would like to have more flexibility and if some informational text at the beginning of the document (like an annotation box) is sufficient, you could do it without any changes to the tool chain by including a file with conditional text.

**Comment by bwgartner** [comment]
So minimally (appearance-wise), the color mapping and the watermark background image would be a great starting point.

Great question WRT to the web-space navigation aspect ... was assuming we might just initially auto-insert the "state" into the title. After reviewing the SBP web page, would it be possible to maybe also insert the color mapping into the row background for the respective document vs. just the alternating white/grey toggle. And then it might make sense to have a "Legend" box at the top to denote the available states and color mapping (or add another column for state labels). Might it also be possible, if a given state has been published (even as earlier, previously-released version), then when the next one is available, provide a short term strikethrough  (and respective colored text) link for the previous one(s), kind of like the "original document archive" with the "state" notation, e.g.

**Foobar** ( State-1 ) formatA formatB formatC ... sorry, couldn't figure out how to do the strikethrough in this space for the "State-1" entry

As for the fixed list of states, I will try to come up with a set for you to review (after I get some other peer feedback as well).

And, of course, conditional text is something I use all the time, but only doing that up front (or even on page headers/footers throughout) is easy to ignore and/or forget while in the depth of the document (compared to color/watermark approach). The intent is to provide awareness for the consumer of the state via repetitive mental pattern image recognition, which could also remind them to revisit to see if another lifecycle stage of the document exists.

**Comment by bwgartner** [comment]
Given our "Draft" and final options, can we first add "Tech Preview" (exactly the same as "Draft" with the backend image or sidebar designation) to create initial content during that stage of our product offerings.

Bonus points to maybe also change the default color-coding of our stylesheet for each of these states (to make it even more visibly obvious as well).
